api/v1beta1/activemqartemis_types.go

```diff
@@ -370,6 +370,34 @@ type DeploymentPlanType struct {
 	// Specifies the Revision History Limit of the statefulset
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Revision History Limit"
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"`
+	// Specifies External PersistentVolumeClaims for the broker pod
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="External PVCs"
+	ExternalPVCs []ExternalPVCType `json:"externalPvcs,omitempty"`
```

Comment:
The array of `ExternalPVCType` could be replaced by a map witht the name as key and the `PersistentVolumeClaimSpec` as value to avoid the new struct type `ExternalPVCType`. Eventually this map could be added to the `ActiveMQArtemisSpec` struct with the name `persistentVolumeClaims`, i.e.
```
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata: 
  name: artemis-role-map
spec:
  persistentVolumeClaims (map[string]corev1.PersistentVolumeClaimSpec):
    myclaim:
      accessModes:
        - ReadWriteOnce
      volumeMode: Filesystem
      resources:
        requests:
          storage: 8Gi
        ...
```

Reply:
ok, will update. Thanks!

Reply:
I have add an option outside the persistentVolumeClaim to allow user to specify mountPath.

Reply:
> I have add an option outside the persistentVolumeClaim to allow user to specify mountPath.

Users could use extraVolumeMounts to define mountPath and other mount options for a persistent volume claim, i.e.
```
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata: 
  name: artemis-role-map
spec:
  persistentVolumeClaims (map[string]corev1.PersistentVolumeClaimSpec):
    myclaim:
      accessModes:
        - ReadWriteOnce
      volumeMode: Filesystem
      resources:
        requests:
          storage: 8Gi
    extraVolumes ([]corev1.Volume):
      - name: mypd
        persistentVolumeClaim:
          claimName: myclaim
    extraVolumeMounts ([]corev1.VolumeMount):
      - mountPath: "/var/www/html"
        name: mypd
        ...
```

Reply:
That shouldn't work. myclaim is a template, it will be translated to real name for each pod, .i.e myclaim-<cr name>-ss-0 for pod0 and so on. extraVolumes and extraVolumeMounts should referencing static volume names, and they are going to mount to all pods. (not one for each pod)

Reply:
I see that we are talking of 2 different options:
1. one persistent volume claim shared with all broker pods;
2. one persistent volume claim template used to define a persistent volume claim for every broker pods;

Option 1 should work without issues. Do we need this option?
```
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata: 
  name: artemis-role-map
spec:
  persistentVolumeClaims (map[string]corev1.PersistentVolumeClaimSpec):
    myclaim:
      accessModes:
        - ReadWriteOnce
      volumeMode: Filesystem
      resources:
        requests:
          storage: 8Gi
  deploymentPlan:
    extraVolumes ([]corev1.Volume):
      - name: mypd
        persistentVolumeClaim:
          claimName: myclaim
    extraVolumeMounts ([]corev1.VolumeMount):
      - mountPath: "/var/www/html"
        name: mypd
        ...
```

Option 2 could work in the same way as it works in [StatefulSetSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#statefulsetspec-v1-apps). The operator could create a persistent volume claim for every broker pods using the template and link an extra volume mount to its related persistent volume claim template using the name, i.e. adding a new filed `volumeClaimTemplates` (a map with the name as key and the PersistentVolumeClaimSpec as value):
```
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata: 
  name: artemis-role-map
spec:
  volumeClaimTemplates (map[string]corev1.PersistentVolumeClaimSpec):
    myclaim:
      accessModes:
        - ReadWriteOnce
      volumeMode: Filesystem
      resources:
        requests:
          storage: 8Gi
  deploymentPlan:
    extraVolumeMounts ([]corev1.VolumeMount):
      - mountPath: "/var/www/html"
        name: myclaim
        ...
```

Reusing `extraVolumeMounts` for `volumeClaimTemplates` too has 3 benefits:
1.  Avoid new types;
2. Reuse an API that users already know, see [StatefulSetSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#statefulsetspec-v1-apps);
3. Expose all [VolumeMount](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#volumemount-v1-core) fields;

Reply:
I think both options deals with different use cases. We should give user those options. Regarding the templates I think you are right that use VolumeMount type to define the mount options for the template. I'll add that. thanks!